tasks/dist.js

```
@@ -0,0 +1,106 @@
+        })
+        .then(function() {
+            return exec('grunt build', {cwd:tempCheckout});
+        })
+        .then(function() {
+            var taggedPkg = grunt.file.readJSON(tempCheckout + '/package.json');
+            var destName = rockyNameForTag(taggedPkg.version);
+            if (versionedRockyExists(taggedPkg.version)) {
+                grunt.log.warn("skipping file", destName, "Tag", tag, "seems to be named incorrectly.");
+                return false;
+            }
+            return exec('cp ' + tempCheckout + '/' + destName + ' ' + destName);
+        });
+    }
+
+    // actual grunt task
+    var desc = 'Checks out and builds tags from remote repository and copies versioned output to ./build/dist';
+    grunt.registerTask('build-missing-dists', desc, function() {
+        var done = this.async();
+
+        // delete temp folder
+        exec( 'rm -rf ' + tempCheckout)
+        // checkout Rocky.js repo into temp folder
+        .then(function() {
+            var url = grunt.file.readJSON('package.json').repository.url;
+            grunt.verbose.writeln('cloning', url, 'into', tempCheckout);
+            return simpleGit('.').promising("clone", url, tempCheckout);
+        })
+        // checkout gh-pages and copy all missing rocky-x.y.z.js from its dist folder to build/dist
+        // we do this so that we will have gh-pages' dist folder + this working copy's rocky.js
+        .then(function() {
+            return simpleGit(tempCheckout).promising('checkout', 'gh-pages');
+        })
+        .then(function() {
+            // copy all rocky.js files from gh-pages' dist into this working copy's build folder, but
+            // preserve existing files in build/dist (presumably only the one we built from this working copy before)
+            return exec('cp -n ' + tempCheckout + '/dist/*.js build/dist')
+                   .catch(function(){return "this is fine since cp -n returns 1 if there was a file already."});
```

Instead of swallowing an error here (`cp` can fail because of many reasons), why not remove build/dist before?